packages/core/lib/commands/test/index.js

```diff
@@ -113,18 +113,32 @@ const command = {
       {
         option: "--stacktrace-extra",
         description: "Shortcut for --stacktrace --compile-all-debug."
+      },
+      {
+        option: "--grep|-g",
+        description: "Use mocha's \"grep\" option while running tests. This " +
+          "option only runs tests that match the supplied regex/string."
       }
     ],
     allowedGlobalOptions: ["network", "config"]
   },
   run: async function (options) {
     const Config = require("@truffle/config");
-    const {Environment, Develop} = require("@truffle/environment");
-    const {copyArtifactsToTempDir} = require("./copyArtifactsToTempDir");
-    const {determineTestFilesToRun} = require("./determineTestFilesToRun");
-    const {prepareConfigAndRunTests} = require("./prepareConfigAndRunTests");
+    const { Environment, Develop } = require("@truffle/environment");
+    const { copyArtifactsToTempDir } = require("./copyArtifactsToTempDir");
+    const { determineTestFilesToRun } = require("./determineTestFilesToRun");
+    const { prepareConfigAndRunTests } = require("./prepareConfigAndRunTests");
+
+    // parse out command line flags to merge in to the config
+    const grep = options.grep || options.g;
+    const bail = options.bail || options.b;
 
-    const config = Config.detect(options);
+    const config = Config.detect(options).merge({
+      mocha: {
+        grep,
+        bail
```

Comment:
Really we can add any mocha options we want. It will just make the docs more junked up the more we add. Can we consider adding other options to be outside the scope of this particular PR?

Reply:
My reasoning is that anyone using mocha filters will have a better experience with the full set of filter options - specify a search by string, or by regex and the ability to invert the search match. Can we vote on this one? Either way I want get this PR approved sooner rather than later.

Reply:
I'm absolutely not opposed to adding more mocha options here. Let's implement them elsewhere.